### ISSUE TITLE
Your loss Implementation might be wrong

### ISSUE THREAD

**Issue by NaiL23** [VinhLoiIT/signet-pytorch#2]
You implement `loss = self.alpha * (1-y) * distance**2 + \
               self.beta * y * (torch.max(torch.zeros_like(distance), self.margin - distance)**2)
       ` as your contrastive loss, however, in your dataset split and preprocessing script, you label (genuine, genuine) as 1 and (genuine,forged) as 0, which means when y=0, your loss = alpha * distance between pairs and it will be minimized, but hopefully they should be as far as possible.

 `loss = self.alpha * y * distance**2 + \
               self.beta * (1-y) * (torch.max(torch.zeros_like(distance), self.margin - distance)**2)
       ` would be a correct implementation.

**Comment by VinhLoiIT** [comment]
It makes sense. Actually, the (assumed) author implemented it [here](https://github.com/sounakdey/SigNet/blob/a555011b72921c020fe72c228c54742b9c364da4/SigNet_v1.py#L52), but I don't remember why it is different in my code.

Thanks for pointing it out.

**Comment by VinhLoiIT** [comment]
As I read the paper, the loss function in the current source code is the same as the formulas (1) in the paper. However, the author said "y is a binary indicator function denoting whether the two samples belong to the same class or not" that it is quite ambiguous
